pallets/providers/src/utils.rs

```diff
@@ -1086,6 +1092,15 @@ where
                 }
             }
             RateDeltaParam::Increase(delta) => {
+                // Get the current bucket rate, which is the rate of a zero sized bucket plus the rate according to the bucket size.
+                let bucket_rate = value_prop
+                    .price_per_unit_of_data_per_block
+                    .checked_mul(&bucket.size.into())
+                    .ok_or(ArithmeticError::Overflow)?
+                    .checked_add(&T::ZeroSizeBucketFixedRate::get())
+                    .ok_or(ArithmeticError::Overflow)?;
```

Comment:
Yeah this is a bit confusing, because here we calculate the `bucket_rate` as:
```rust
                let bucket_rate = value_prop
                    .price_per_unit_of_data_per_block
                    .checked_mul(&bucket.size.into())
                    .ok_or(ArithmeticError::Overflow)?
                    .checked_add(&T::ZeroSizeBucketFixedRate::get())
                    .ok_or(ArithmeticError::Overflow)?;
```

But in `RateDeltaParam::NewBucket` or `RateDeltaParam::RemoveBucket` we calculate it as:
```rust
                let bucket_rate = value_prop
                    .price_per_unit_of_data_per_block
                    .checked_mul(&bucket.size.into())
                    .ok_or(ArithmeticError::Overflow)?;
```

And only then add the zero size bucket rate to the `new_rate`. We could unify it. But no big deal.

Reply:
I unified this logic using `multiply_rational` so it should be consistent and more concise now